lib/mapper.js

```diff
@@ -22,13 +52,52 @@ exports.track = function(track){
     objCase.del(props, 'revenue');
   }
 
-  return {
-    email: track.email(),
-    userId: track.userId(),
-    eventName: track.event(),
-    createdAt: unixTime(track.timestamp()),
-    dataFields: props
-  };
+  var eventName = track.event();
+  var cart = props["products"] || props["cart"];
+  if (cart && (eventName == "Added Product" || eventName == "Removed Product")) {
+    return {
+      user: {
+        email: track.email(),
+        userId: track.userId()
+      },
+      items: cartToIterableCart(cart),
+      isUpdateCart: true
+    };
+  } else if (cart && eventName == "Completed Order") {
+    var total = track.total();
+    if (!total) {
```

Comment:
total is a spec'd field — it should always be present in customers' `Completed Order` calls